test/floatfuncs.jl

```diff
@@ -68,3 +68,5 @@ for elty in (Float32,Float64)
         @test round(elty2,A) == fill(round(elty2,x),(10,10,10))
     end
 end
+
+@test float(Char) == Any
```

Comment:
That may be a vestige from when `Char` was a type of number. It isn't anymore, so we should delete this definition.